codec-http/src/test/java/io/netty/handler/codec/http/HttpContentDecoderTest.java

```diff
@@ -218,6 +227,64 @@ public void testExpectContinueResponse4() {
     }
 
     @Test
+    public void testExpectContinueResetHttpObjectDecoder() {
+        // request with header "Expect: 100-continue" must be replied with one "100 Continue" response
+        // case 5: Test that HttpObjectDecoder correctly resets its internal state after a failed expectation.
+        HttpRequestDecoder decoder = new HttpRequestDecoder();
+        final int maxBytes = 10;
+        HttpObjectAggregator aggregator = new HttpObjectAggregator(maxBytes);
+        final AtomicReference<FullHttpRequest> secondRequestRef = new AtomicReference<FullHttpRequest>();
+        EmbeddedChannel channel = new EmbeddedChannel(decoder, aggregator, new ChannelInboundHandlerAdapter() {
+            @Override
+            public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+                if (msg instanceof FullHttpRequest) {
+                    if (!secondRequestRef.compareAndSet(null, (FullHttpRequest) msg)) {
+                        ((FullHttpRequest) msg).release();
+                    }
+                } else {
+                    ReferenceCountUtil.release(msg);
+                }
+            }
+        });
+        String req1 = "POST /1 HTTP/1.1\r\n" +
+                "Content-Length: " + (maxBytes + 1) + "\r\n" +
+                "Expect: 100-continue\r\n" +
+                "\r\n";
+        assertFalse(channel.writeInbound(Unpooled.wrappedBuffer(req1.getBytes())));
+
+        Object o = channel.readOutbound();
```

Comment:
@Scottmitch just directly use `FullHttpResponse resp = ...`. This will fail the test if you receive something unexpected and so you also not need the `assertThat(...)`